### PR TITLE
feat: add test for grafana ingress

### PR DESCRIPTION
**Pull request by Alex-Vovchuk**

### CODE REVIEW

**Review by gmlexx — CHANGES_REQUESTED** [review]
Check comments, please, also some copilot comments looks reasonable

File: Makefile [hunk 1]
```diff
@@ -484,6 +484,40 @@ support-bundle: envsubst support-bundle-cli
...
+	"$$HELM_BIN" upgrade --install ingress-nginx ingress-nginx/ingress-nginx \
+		-n ingress-nginx \
+		--create-namespace \
+		--wait \
+		--timeout 10m && \
+	kubectl -n ingress-nginx rollout status deploy/ingress-nginx-controller --timeout=10m
+
+.PHONY: dev-grafana-ingress-smoke-test
+dev-grafana-ingress-smoke-test:
+	kubectl -n ingress-nginx port-forward svc/ingress-nginx-controller 8443:443 >/dev/null 2>&1 &
```
[the remark this answers is not in the document]
Comment (Alex-Vovchuk) [reply Makefile]: Fixed everything from comments.

File: charts/kof/values-local.yaml [hunk 2]
```diff
@@ -112,9 +112,18 @@ kof-mothership:
...
+
+      ingress:
+        enabled: true
+        host: grafana.kof.local
       security:
         create_secret: true
 
+    cert-manager:
+      cluster-issuer:
+        create: true
```
Comment (denis-ryzhkov) [inline charts/kof/values-local.yaml]: Please add explicit `cluster-issuer: create: false` to the prod `values.yaml`, as it does not have these values [yet](https://github.com/k0rdent/kof/blob/main/charts/kof/values.yaml#L133-L134) and they are required in the `(index . "cluster-issuer" "create")` above in this PR.

**Review by denis-ryzhkov — CHANGES_REQUESTED** [review]

**Review by denis-ryzhkov — DISMISSED** [review]

File: charts/kof/values-local.yaml [hunk 3]
```diff
@@ -117,10 +117,29 @@ kof-mothership:
...
         create_secret: true
       gateway:
         enabled: true
+        httpRoute:
+          spec:
+            parentRefs:
+              - name: gateway
+                namespace: kof
+            hostnames:
+              - grafana.kof.local
```
Comment (gmlexx) [inline charts/kof/values-local.yaml:127]: Other httpRoute values are default, no need to specify them

File: charts/kof-mothership/templates/cert-manager/clusterissuer.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+{{- if eq (index . "cluster-issuer" "provider") "letsencrypt" }}
+  acme:
+    server: https://acme-v02.api.letsencrypt.org/directory
+    email: {{ .email }}
+    privateKeySecretRef:
+      name: letsencrypt-prod
+    solvers:
+      - http01:
+          ingress:
+            ingressClassName: nginx
```
Comment (gmlexx) [inline charts/kof-mothership/templates/cert-manager/clusterissuer.yaml]: Need to rebase from https://github.com/k0rdent/kof/tree/mothership-gateway with solvers for gateway

File: charts/kof/values-local.yaml [hunk 5]
```diff
@@ -174,6 +195,17 @@ kof-regional:
...
+      enabled: true
+    gateway:
+      enabled: true
+      annotations: null
+      spec:
+        gatewayClassName: eg
+        listeners:
+          - name: http
+            port: 80
+            protocol: HTTP
```
Comment (gmlexx) [inline charts/kof/values-local.yaml]: Please remove, we don't need gateway for local storage

File: charts/kof/values-local.yaml [hunk 6]
```diff
@@ -148,9 +160,18 @@ kof-regional:
         cluster-issuer:
           provider: self-signed
           name: selfsigned
+      gatewayClass:
+        enabled: true
```
Comment (gmlexx) [inline charts/kof/values-local.yaml]: Please remove, enabled in MCS

File: charts/kof/values-local.yaml [hunk 7]
```diff
@@ -148,9 +160,18 @@ kof-regional:
...
       gateway:
+        enabled: true
```
Comment (gmlexx) [inline charts/kof/values-local.yaml]: Please remove, enabled in MCS

File: charts/kof/values-local.yaml [hunk 8]
```diff
@@ -148,9 +160,18 @@ kof-regional:
...
         annotations:
           cert-manager.io/cluster-issuer: selfsigned
+        spec:
+          gatewayClassName: eg
+          listeners:
+            - name: http
+              port: 80
+              protocol: HTTP
```
Comment (gmlexx) [inline charts/kof/values-local.yaml]: Please remove, enabled in MCS

**Review by gmlexx — CHANGES_REQUESTED** [review]

File: .github/workflows/pr_test_adopted_upgrade.yml [hunk 9]
```diff
@@ -155,9 +155,24 @@ jobs:
...
+          kubectl -n envoy-gateway-system rollout status deploy/envoy-gateway --timeout=10m
       - name: "[Latest KOF Release] Deploy KOF Components"
         run: |
           DISABLE_KOF_COLLECTORS=true DISABLE_KOF_STORAGE=true make dev-deploy
+      - name: "[Latest KOF Release] Pin adopted-cluster template to 1-0-1"
+        # KCM v1.8.0-release ships only adopted-cluster-1-0-1; adopted-cluster-1-0-2
+        # becomes available after the KCM upgrade to v1.9.0-release later in this job.
+        # Fetch PR Ref and Checkout Repo restores 1-0-2 for the PR-branch upgrade step.
+        run: |
+          sed -i 's/template: adopted-cluster-1-0-2/template: adopted-cluster-1-0-1/g' demo/cluster/adopted-cluster-*.yaml
```
Comment (gmlexx) [inline .github/workflows/pr_test_adopted_upgrade.yml]: Looks like this should not be needed

File: charts/kof/values-local.yaml (hunk 7 above)
Comment (Alex-Vovchuk) [reply charts/kof/values-local.yaml]: Deleted

File: charts/kof/values-local.yaml (hunk 8 above)
Comment (Alex-Vovchuk) [reply charts/kof/values-local.yaml]: Deleted

File: charts/kof/values-local.yaml (hunk 6 above)
Comment (Alex-Vovchuk) [reply charts/kof/values-local.yaml]: Deleted

File: .github/workflows/pr_test_adopted_upgrade.yml (hunk 9 above)
Comment (Alex-Vovchuk) [reply .github/workflows/pr_test_adopted_upgrade.yml]: Deleted

**Review by gmlexx — DISMISSED** [review]

File: .github/workflows/pr_test_kcm_region_with_kof.yaml [hunk 10]
```diff
@@ -115,6 +115,14 @@ jobs:
       - name: "[PR] Deploy KOF components on Mothership cluster"
         run: |
           DISABLE_KOF_COLLECTORS=true DISABLE_KOF_STORAGE=true make dev-deploy
+      - uses: actions/setup-python@v6
+        with:
+          python-version: "3.13"
+          cache: "pip"
```
Comment (denis-ryzhkov) [inline .github/workflows/pr_test_kcm_region_with_kof.yaml]: As we've disabled cache in other PRs, let's not re-add it here too.
If this is the only change you may apply it in another PR:

```suggestion
```

File: charts/kof/values-local.yaml [hunk 11]
```diff
@@ -117,10 +117,41 @@ kof-mothership:
     grafana:
       installForTests: true
       enabled: true
+
+      ingress:
+        enabled: false
```
Comment (denis-ryzhkov) [inline charts/kof/values-local.yaml]: It's already false in the default [values.yaml](https://github.com/k0rdent/kof/blob/75d70fc30581ef73e911119e076e602b9dc274ae/charts/kof/values.yaml#L199) so let's keep `values-local.yaml` a minimal patch:

```suggestion
```

File: .github/workflows/pr_test_kcm_region_with_kof.yaml [hunk 12]
```diff
@@ -115,6 +115,14 @@ jobs:
...
+      - run: pip install -r scripts/requirements.txt
+      - name: "[PR] Grafana gateway smoke test"
+        run: |
+          make dev-grafana-smoke
```
Comment (denis-ryzhkov) [inline .github/workflows/pr_test_kcm_region_with_kof.yaml:124]: So we run grafana smoke test only in one of our workflows - the one testing KCM region.

What if grafana fails in other use cases/workflows?

I guess we should add this smoke test there too.

File: scripts/grafana_gateway_smoke_test.py [hunk 13]
```diff
@@ -0,0 +1,101 @@
...
+        except AssertionError as exc:
+            if time.monotonic() >= deadline:
+                pytest.fail(f"Timed out ({timeout}s) waiting for {description}: {exc}")
+            print(f"  [{deadline - time.monotonic():.0f}s] waiting for {description}...")
+            time.sleep(POLL)
+
+# Setup fixture (runs once per test session)
+
+@pytest.fixture(scope="module", autouse=True)
+def setup_tls() -> None:
```
Comment (denis-ryzhkov) [inline scripts/grafana_gateway_smoke_test.py:40]: Looks like you're testing cert and gateway in a module autouse fixture to prevent other tests from running before we have gateway in place, right?

But in this case there is no need to test them again later, so please delete `test_certificate_issued` and `test_gateway_programmed` which duplicate this fixture.

**Review by denis-ryzhkov — CHANGES_REQUESTED** [review]

**Review by denis-ryzhkov — APPROVED** [review]
